src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -243,8 +243,8 @@ namespace System
     internal static partial class Number
     {
         internal const int DecimalPrecision = 29; // Decimal.DecCalc also uses this value
-        private const int FloatPrecision = 7;
-        private const int DoublePrecision = 15;
```

Comment:
This changes the default for all format specifiers which results in "less pretty", but more accurate numbers for the default `double.ToString()` case.

As an example, see the below table:

Literal Value | ToString Before | ToString After
------------- | --------------- | --------------
0.0           | 0               | 0
0.1           | 0.1             | 0.10000000000000001
0.2           | 0.2             | 0.20000000000000001
0.3           | 0.3             | 0.29999999999999999
0.4           | 0.4             | 0.40000000000000002
0.5           | 0.5             | 0.5
0.6           | 0.6             | 0.59999999999999998
0.7           | 0.7             | 0.69999999999999996
0.8           | 0.8             | 0.80000000000000004
0.9           | 0.9             | 0.90000000000000002
1.0           | 1               | 1

Where `Literal Value` is the value given to C# source code

Reply:
I do not think that this is a good change.

Reply:
We could:
* Keep things as I have in the PR, where the default for all format specifiers is the "correct" value.
  * This results in less pretty, but more accurate numbers.
* Keep the existing default precision and digits (`15` and `6`) for everything that isn't `"R"`
  * This results in correct round-tripping behavior, and the same behavior for everything else
* Keep the new default precision, but change the default digits to be the old precision specifier.
  * This will keep most strings for the default specifiers the same, but will produce different/more accurate results it a few cases where that might matter
    * This should only impact numbers where the 16th or 17th digit impacted rounding of the 15th